src/components/Dialog/useResizeObserver.js

```diff
@@ -0,0 +1,25 @@
+import { useRef, useLayoutEffect, useState, useCallback } from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+
+// https://tobbelindstrom.com/blog/resize-observer-hook/
+const useResizeObserver = () => {
+  const [observerEntry, setObserverEntry] = useState({});
+  const [node, setNode] = useState(null);
+  const observer = useRef(null);
+
+  const disconnect = useCallback(() => observer.current?.disconnect(), []);
+
+  const observe = useCallback(() => {
+    observer.current = new ResizeObserver(([entry]) => setObserverEntry(entry));
+    if (node) observer.current.observe(node);
+  }, [node]);
+
+  useLayoutEffect(() => {
+    observe();
+    return () => disconnect();
+  }, [disconnect, observe]);
+
+  return [setNode, observerEntry];
+};
+
+export default useResizeObserver;
```

Comment:
Technically I want to just download a hook for this, but all of them didn't work. That being said, it's likely they didn't work due to a false negative I discovered (and fixed) later working on this component. I'll need to revisit it but not incredibly pressing.